libraries/AP_Scripting/drivers/TOFSense-M/TOFSense-M_Serial.lua

```diff
@@ -0,0 +1,252 @@
+local update_rate_ms    = 10    -- update rate (in ms) of the driver. 10ms was found to be appropriate
+local bytes_to_parse    = 100   -- serial bytes to parse in one interation of lua script. Reduce this if script is not able to complete in time
+local debug_enable      = false -- helpgul debug GCS prints
+
+-- Global variables (DO NOT CHANGE)
+local NOOPLOOP_FRAME_HEADER = 0x57
+local NOOPLOOP_FRAME_HEADER_1 = 0x01
+local param_num_lua_driver_backend = 36         -- parameter number for lua rangefinder
+local param_num_lua_prx_backend = 15            -- parameter number for lua proximity
+local lua_driver_backend                        -- store lua backend here
+local sensor_driver_found = false               -- true if user has configured lua backend
+local num_pixels = 16                           -- automatically updated if 64. User can select between 16/64 from NAssistant software
+local total_bytes_to_expect = 112               -- total bytes in one complete packet
+local linebuf = {}                              -- serial buffer
+local linebuf_len = 0
+local index_row_max = 8
+
+local PARAM_TABLE_KEY = 109 + sensor_no
+local PARAM_TABLE_PREFIX = string.format("TOFSENSE_S" .. sensor_no.. "_")
+
+-- bind a parameter to a variable
+function bind_param(name)
+   local p = Parameter()
+   assert(p:init(name), string.format('could not find %s parameter', name))
+   return p
+end
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+   assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+   return bind_param(PARAM_TABLE_PREFIX .. name)
+end
+
+-- setup parameters
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 3), 'could not add param table')
+
+--[[
+  // @Param: TOFSENSE_S1_PRX
+  // @DisplayName: TOFSENSE-M to be used as Proximity sensor
+  // @Description: Set 0 if sensor is to be used as a 1-D rangefinder (minimum of all distances will be sent, typically used for height detection). Set 1 if it should be used as a 3-D proximity device (Eg. Obstacle Avoidance)
+  // @Values: 0:Set as Rangefinder, 1:Set as Proximity sensor
+  // @User: Standard
+--]]
+SET_PRX    = bind_add_param('PRX', 1, 0)
+
+--[[
+  // @Param: TOFSENSE_S1_SP
+  // @DisplayName: TOFSENSE-M serial port config
+  // @Description: UART instance sensor is connected to. Set 1 if sensor is connected to the port with fist SERIALx_PROTOCOL = 28. 
+  // @Range: 1 4
+  // @User: Standard
+--]]
+SERIAL_PORT = bind_add_param('SP', 2, 1)
+
+--[[
+  // @Param: TOFSENSE_S1_BR
+  // @DisplayName: TOFSENSE-M serial port baudrate
+  // @Description: Serial Port baud rate. Sensor baud rate can be changed from Nassistant software
+  // @User: Standard
+--]]
+SERIAL_BAUD = bind_add_param('BR', 3, 230400)
```

Comment:
I think we should stick with the existing SERIALx_BAUD instead of creating new parameter for it.

Reply:
I tried removing the port:begin(BAUD_RATE) call in the script, thinking it would take the SERIALx_BAUD automatically. It did work in SITL, but unfortunately, it did not work on a Cube Orange.